src/Sylius/Bundle/CoreBundle/ExpressionLanguage/ShopperContextVariables.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\ExpressionLanguage;
+
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Core\Context\ShopperContextInterface;
+use Sylius\Component\Customer\Model\CustomerInterface;
+use Sylius\Resource\Symfony\ExpressionLanguage\VariablesInterface;
+
+final readonly class ShopperContextVariables implements VariablesInterface
+{
+    public function __construct(
+        private ShopperContextInterface $shopperContext,
+    ) {
+    }
+
+    /**
+     * @return array{
+     *     sylius_current_channel: ChannelInterface,
+     *     sylius_current_customer: CustomerInterface,
+     *     sylius_current_locale_code: string,
+     *     sylius_shopper_context: ShopperContextInterface,
+     * }
+     */
+    public function getVariables(): array
+    {
+        return [
+            'sylius_current_channel' => $this->shopperContext->getChannel(),
+            'sylius_current_customer' => $this->shopperContext->getCustomer(),
+            'sylius_current_locale_code' => $this->shopperContext->getLocaleCode(),
```

Comment:
`_current_` sounds fine, but here is possible alternative.

```suggestion
            'sylius_contextual_channel' => $this->shopperContext->getChannel(),
            'sylius_contextual_customer' => $this->shopperContext->getCustomer(),
            'sylius_contextual_locale_code' => $this->shopperContext->getLocaleCode(),
```